src/java.base/share/classes/java/nio/charset/Charset.java

```diff
@@ -619,7 +618,14 @@ public static SortedMap<String,Charset> availableCharsets() {
         return Collections.unmodifiableSortedMap(m);
     }
 
-    private @Stable static Charset defaultCharset;
+    private static final Supplier<Charset> defaultCharset = StableValue.supplier(
+            new Supplier<>() { public Charset get() { return defaultCharset0(); }});
+
+    private static Charset defaultCharset0() {
+        return Objects.requireNonNullElse(
+                standardProvider.charsetForName(StaticProperty.fileEncoding()),
+                sun.nio.cs.UTF_8.INSTANCE);
```

Comment:
I think it would more readable to keep the if-then-else rather than using requireNonNullElse here.